controllers/manila_controller.go

```diff
@@ -1126,3 +1136,45 @@ func (r *ManilaReconciler) ensureDB(
 	instance.Status.Conditions.MarkTrue(condition.DBReadyCondition, condition.DBReadyMessage)
 	return db, ctrlResult, nil
 }
+
+// checkManilaSubCRGeneration -
+func (r *ManilaReconciler) checkManilaSubCRGeneration(
```

Comment:
This seems like 3 function merged together with the subCR's type erased. I would just have 3 independent functions.

Reply:
The goal of a single function is to minimize the duplicate code.
Is your point related to the fact that this function is not clear, or should it be different (like using generics)? 
Or do you see other problems related to the fact I have an `interface{}` in place of the subCR explicit type for each of them?

Reply:
As far as I see the duplicated code it reduces is only
```
listOpts := []client.ListOption{
		client.InNamespace(instance.Namespace),
	}
```
the rest of the current function is different for each type the function handles. 
So I don't think this function is really reduces duplication.

Reply:
while technically the single function would work, I think having three separate functions is ok because improves readability and it helps in terms of maintenance, so I switched the implementation to three functions I originally had in the first commit.